src/lazy/binary/raw/value.rs

```diff
@@ -38,9 +41,29 @@ impl<'a> Debug for LazyRawValue<'a> {
     }
 }
 
-type ValueParseResult<'a> = IonResult<RawValueRef<'a>>;
+type ValueParseResult<'data, F> = IonResult<RawValueRef<'data, F>>;
 
-impl<'data> LazyRawValue<'data> {
+impl<'data> LazyRawValuePrivate<'data> for LazyRawBinaryValue<'data> {
+    fn field_name(&self) -> Option<RawSymbolTokenRef<'data>> {
+        self.encoded_value.field_id.map(RawSymbolTokenRef::SymbolId)
+    }
+}
+
+impl<'data> LazyRawValue<'data, BinaryFormat> for LazyRawBinaryValue<'data> {
+    fn ion_type(&self) -> IonType {
+        self.ion_type()
+    }
+
+    fn annotations(&self) -> RawBinaryAnnotationsIterator<'data> {
+        self.annotations()
+    }
+
+    fn read(&self) -> IonResult<RawValueRef<'data, BinaryFormat>> {
+        self.read()
+    }
```

Comment:
🗺️ These impls just forward the method call to the inherent implementation on the type.